hack/observability/kube-state-metrics/crd-config.yaml

```diff
@@ -0,0 +1,745 @@
+kind: CustomResourceStateMetrics
+spec:
+  resources:
+  - groupVersionKind:
+      group: cluster.x-k8s.io
+      kind: Cluster
+      version: v1beta1
+    labelsFromPath:
+      name:
+      - metadata
+      - name
+      namespace:
+      - metadata
+      - namespace
+      uid:
+      - metadata
+      - uid
+    metricNamePrefix: capi_cluster
+    metrics:
+    - name: info
+      help: Information.
```

Comment:
When this PR merge let's open a follow up issue to improve metrics descriptions (most of them are basic or TODO)

Reply:
ah, missed that and wanted to add help everywhere :-( will update it :-)

Reply:
Updated all metrics to contain help information. Texts got adapted from existing kube-state-metrics help information.